File: integration/py/delete.py
import sys
from typing import NamedTuple, List

import synnax as sy
from integration import FILE_NAME

class TestConfig(NamedTuple):
    identifier: str
    expected_error: str
    channels: List[str]
    time_range: sy.TimeRange


client = sy.Synnax(
    host="localhost",
    port=9090,
    username="synnax",
    password="seldon",
    secure=False,
)


class Delete_Test:
    _tc: TestConfig
    def __init__(self, argv: List[str]):
        argv_counter = 1
        identifier = argv[argv_counter]
        argv_counter += 1
        time_range_start = int(argv[argv_counter])
        argv_counter += 1
        time_range_end = int(argv[argv_counter])
        argv_counter += 1
        expected_error = argv[argv_counter]
        argv_counter += 1
        num_channels = int(argv[argv_counter])
        argv_counter += 1
        channels = []
        for _ in range(num_channels):
            channels.append(argv[argv_counter])
            argv_counter += 1

        self._tc = TestConfig(
            identifier=identifier,
            time_range=sy.TimeRange(sy.TimeStamp(time_range_start), sy.TimeStamp(time_range_end)),
            expected_error=expected_error,
            channels=channels,
        )


    def test(self):
        client.delete(self._tc.channels, self._tc.time_range)


    def test_with_timing(self):
        start = sy.TimeStamp.now()
        error_assertion_passed = False
        actual_error = "no_error"
        try:
            self.test()
        except Exception as e:
            actual_error = str(e)
            if self._tc.expected_error != "no_error" and self._tc.expected_error in str(e):
                error_assertion_passed = True
            else:
                raise(e)
        else:
            actual_error = "no_error"
            if self._tc.expected_error == "no_error":
                error_assertion_passed = True
        end = sy.TimeStamp.now()

        s = f'''
-- Python Delete ({self._tc.identifier})--
Time taken: {start.span(end)}
Configuration:
\tNumber of channels: {len(self._tc.channels)}
Expected error: {self._tc.expected_error}; Actual error: {actual_error}\n{"PASS!!" if error_assertion_passed else "FAIL!!!!"}
'''
        with open(FILE_NAME, "a") as f:
            f.write(s)


def main():
    Delete_Test(sys.argv).test_with_timing()


if __name__ == "__main__":
    main()

File: integration/ts/src/delete.ts
// Copyright 2025 Synnax Labs, Inc.
//
// Use of this software is governed by the Business Source License included in the file
// licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with the Business Source
// License, use of this software will be governed by the Apache License, Version 2.0,
// included in the file licenses/APL.txt.

import { Synnax } from "@synnaxlabs/client";
import { TimeRange, type TimeSpan, TimeStamp } from "@synnaxlabs/x";
import * as fs from "fs";
import { argv } from "process";

class TestConfig {
  identifier: string = "";
  expectedError: string = "";
  channels: string[] = [];
  timeRange: TimeRange = TimeRange.ZERO;
}

const client = new Synnax({
  host: "localhost",
  port: 9090,
  username: "synnax",
  password: "seldon",
  secure: false,
});

class DeleteTest {
  tc: TestConfig;
  constructor(argv: string[]) {
    let argvCounter = 2;
    const identifier = argv[argvCounter++];
    const timeRangeStart = BigInt(argv[argvCounter++]);
    const timeRangeEnd = BigInt(argv[argvCounter++]);
    const expectedError = argv[argvCounter++];
    const number_of_channels = parseInt(argv[argvCounter++]);
    const channels = [];
    for (let i = 0; i < number_of_channels; i++) channels.push(argv[argvCounter++]);

    this.tc = {
      identifier,
      expectedError,
      timeRange: new TimeRange(timeRangeStart, timeRangeEnd),
      channels,
    };
  }

  async testWithTiming(): Promise<void> {
    const start = TimeStamp.now();
    let errorAssertion = false;
    let actualError = "";
    let caught = false;
    await this.test().catch((e: unknown) => {
      console.log("CAUGHT: ", e);
      if (e instanceof Error) {
        caught = true;
        actualError = e.message;
        if (
          this.tc.expectedError != "no_error" &&
          e.message.includes(this.tc.expectedError)
        )
          errorAssertion = true;
        else throw e;
      } else throw e;
    });
    if (!caught) {
      if (this.tc.expectedError == "no_error") errorAssertion = true;
      actualError = "no_error";
    }
    const end = TimeStamp.now();

    const time: TimeSpan = start.span(end);
    const s = `
-- TypeScript Delete (${this.tc.identifier}) --
Time taken: ${time.isZero ? 0 : time}
Configuration:
\tNumber of channels: ${this.tc.channels.length}

Expected error: ${this.tc.expectedError}; Actual error: ${actualError}: ${errorAssertion ? "PASS!!" : "FAIL!!!!"}
`;

    fs.appendFileSync("../../timing.log", s);
  }

  async test(): Promise<void> {
    client.delete(this.tc.channels, this.tc.timeRange);
  }
}

async function main() {
  try {
    await new DeleteTest(argv).testWithTiming();
  } finally {
    client.close();
  }
}

await main();
